extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -236,11 +263,21 @@ export class Visibility {
    * @param {!VisibilityListenerCallbackDef} callback
    * @param {boolean} shouldBeVisible True if the element should be visible
    *  when callback is called. False otherwise.
+   * @param {Element} analyticsElement The amp-analytics element that the
+   *  config is associated with.
    */
-  listenOnce(config, callback, shouldBeVisible) {
-    const element = this.win_.document.getElementById(config['selector']
-        .slice(1));
+  listenOnce(config, callback, shouldBeVisible, analyticsElement) {
+    const selector = config['selector'];
+    const element = getElement(selector, analyticsElement,
+        config['selectionMethod']);
+    if (!element) {
+      user().error('Element not found for visibilitySpec: ' + selector);
+    }
     const res = this.resourcesService_.getResourceForElement(element);
+    if (!res) {
```

Comment:
oh. I added a try catch block to catch that and show a more meaningful error. Hope that's okay.